tests/components/homeassistant/triggers/test_time_pattern.py

```diff
@@ -312,6 +312,251 @@ async def test_if_fires_periodic_hours(
     assert len(service_calls) == 1
 
 
+async def test_if_fires_range_seconds(
```

Comment:
Looks like these tests can be parametrized:
- pattern
- fire_time_changed

Reply:
Done. I didn't initially because the initial author didn't, but i agree it's much better this way.